src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the Duke User Interface which deals with interactions with the user
+ */
+public class Ui {
+    public Ui() { }
+
+    /**
+     * Processes DukeException errors and displays them to the user.
+     *
+     * @param e The DukeException thrown
+     */
+    public void showLoadingError(DukeException e) {
+        System.out.println(e.getMessage());
+    }
+
+    public void greeting() {
+        System.out.println("Hello! I'm Duke\n" + "What can I do for you?");
+    }
+
+    /**
+     * Simulates an exit event when the user inputs the bye command.
+     *
+     * @return Returns "Bye" if the user inputs the bye command.
+     */
+    public String goodbye() {
+        System.out.println("Bye! Hope to see you again soon!");
+        return "Bye";
+    }
+
+    /**
+     * Displays the user's current task list.
+     *
+     * @param tasks The current TaskList of the User.
+     */
+    public void list(TaskList tasks) {
```

Comment:
I think the naming of the method is generally okay. But I think that maybe shall we change it to ```displayList(TaskList tasks)``` to make it more detailed and specific? The same applies to the other method namings below.